### PR TITLE
Add a release instruction list for other devs to release

### PR DESCRIPTION
**Pull request by cgruber**

### CODE REVIEW

**Review by rogerhu — APPROVED** [review]

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# Release Instructions.
```
Comment (rogerhu) [inline RELEASE.md]: nitpick - do you need a period for a title?

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+> Note: Versions should be semantic versioning maj.min.patch style, except for the default
+> HEAD-SNAPSHOT which is the version used on the `main` branch
+
+Preconditions:
+
+  1. You must have a username on oss.sonatype.org.
+    - If you need one, sign up at issues.sonatype.org.
```
Comment (rogerhu) [inline RELEASE.md:9]: always thought it was funny you had to sign up for an account via a JIRA ticket

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+             --key <yourgpgkey> \
+             --username <yourusername> \
+             --password <yourpassword> \
+         ```
+          - username/password are for oss.sonatype.org
+      2. Enter your gnupg passphrase for the given PGP key (if you set a passphrase for it).
+      3. Wait for the deployment to push all the files to sonatype. 
+      4. Log in to `oss.sonatype.org`
+      5. Search for "staging repositories"
+      6. Find a "comsquareup-<somenumber>" repo.
```
Comment (rogerhu) [inline RELEASE.md]: comsquareup is right right?

Comment (cgruber) [reply RELEASE.md]: it is right. The repo names are temporary, and it's just a mnemonic in their UI, so it's just the first bits of the groupID squashed together.

File: RELEASE.md (hunk 2 above)
Comment (cgruber) [reply RELEASE.md:9]: ***twitches***

File: RELEASE.md (hunk 1 above)
Comment (cgruber) [reply RELEASE.md]: Nope - good catch.

File: RELEASE.md (hunk 3 above)
Comment (cgruber) [reply RELEASE.md]: But... that needs backticks, not quotes, so `<somenumber>` isn't treated as html.

**Pull request merged**